fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanApplicationWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -794,19 +794,25 @@ public CommandProcessingResult rejectApplication(final Long loanId, final JsonCo
 
         final AppUser currentUser = getAppUserIfPresent();
 
-        this.loanApplicationTransitionApiJsonValidator.validateRejection(command.json());
+        // validate before loan assembling
+        loanApplicationTransitionValidator.validateRejection(command.json());
 
+        // retrieve/assemble loan
         final Loan loan = retrieveLoanBy(loanId);
 
+        // check client or group
         checkClientOrGroupActive(loan);
 
+        // check for mandatory entities
         entityDatatableChecksWritePlatformService.runTheCheckForProduct(loanId, EntityTables.LOAN.getName(),
                 StatusEnum.REJECTED.getCode().longValue(), EntityTables.LOAN.getForeignKeyColumnNameOnDatatable(), loan.productId());
 
+        // validate loan rejection
+        loanApplicationTransitionValidator.validateRejection(loan, defaultLoanLifecycleStateMachine, command);
```

Comment:
Please merge together the loanApplicationTransitionValidator.validateRejection methods. I dont think we need 2.

Reply:
@adamsaghy  second validation is on loan while the first validation is on json. hence kept in two methods.

Reply:
We can do all the validations before executing the rejection. You can load the loan in the first validation method as well if needed. ;)

Reply:
done